Tools/wasm/emscripten/__main__.py

```diff
@@ -215,12 +215,29 @@ def configure_emscripten_python(context, working_dir):
     exec_script = working_dir / "python.sh"
     exec_script.write_text(
         dedent(
-            f"""\
+            """\
             #!/bin/sh
 
+            # Macs come with a defective fork of coreutils so feature detect and
+            # work around it.
+            if which grealpath > /dev/null; then
+                # It has brew installed gnu core utils, use that
+                REALPATH="grealpath -s"
+            elif which readlink > /dev/null && realpath --version | grep GNU > /dev/null; then
+                # realpath points to GNU realpath so use it.
+                REALPATH="realpath -s"
+            else
+                # Shim for macs without GNU coreutils
+                abs_path () {
+                    echo "$(cd $(dirname "$1");pwd)/$(basename "$2")"
+                }
+                REALPATH=abs_path
+            fi
+            """
+            f"""\
```

Comment:
The `\` here leads to some odd indentation of the first comment line.
```suggestion
            f"""
```